presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java

```diff
@@ -191,12 +196,15 @@ else if (inDictionary[i]) {
         return block;
     }
 
-    private void setDictionaryBlockData(Slice[] dictionary)
+    private void setDictionaryBlockData(byte[] dictionaryData, int[] dictionaryOffsets, int positionCount)
     {
+        verify(positionCount > 0);
         // only update the block if the array changed to prevent creation of new Block objects, since
         // the engine currently uses identity equality to test if dictionaries are the same
-        if (dictionaryBlock.getValues() != dictionary) {
-            dictionaryBlock = new SliceArrayBlock(dictionary.length, dictionary, true);
+        if (dictionaryBlock.getSlice().getBase() != dictionaryData) {
```

Comment:
I'm not a fan of adding a `getSlice` method to block.  Is there any other way to do this?

Reply:
Maybe we can just use another variable to keep tracking of the base; something like:
```
private byte[] currentDictionaryData = EMPTY_DICTIONARY_DATA;
...

if (currentDictionaryData != dictionaryData) {
    ...
    dictionaryBlock = new VariableWidthBlock(positionCount, Slices.wrappedBuffer(dictionaryData), dictionaryOffsets, isNullVector);
    currentDictionaryData = dictionaryData;
}